### PR TITLE
Exception handling and validation implementation

### PR DESCRIPTION
**Pull request by Enessetere**
Add validation to input forms and validation exception handling to User controller.

### CODE REVIEW

File: src/main/java/pl/arcadeit/forex/exception/UserExceptionHandler.java [hunk 1]
```diff
@@ -1,17 +1,49 @@
...
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 @RestControllerAdvice(assignableTypes = UserController.class)
 public class UserExceptionHandler {
 
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ExceptionHandler(Exception.class)
+    public ErrorMessage handleGlobalException(final Exception exception) {
```
Comment (PrzyblowskiBartosz) [inline src/main/java/pl/arcadeit/forex/exception/UserExceptionHandler.java]: proposal for the future: separate handleGlobalException to other class GlobalExceptionHandler

**Review by PrzyblowskiBartosz — DISMISSED** [review]
All changes LGFM

File: src/main/java/pl/arcadeit/forex/domain/User.java [hunk 2]
```diff
@@ -22,22 +23,27 @@
 public class User {
 
     @Id
-    @Email(message = "Email address is invalid.")
+    @NotNull(message = "{email.required}")
+    @Email(message = "{email.invalid}")
```
Comment (BartlomiejBak) [inline src/main/java/pl/arcadeit/forex/domain/User.java:27]: Why there are two conventions of messages ("{message}" and "message")?

File: src/main/java/pl/arcadeit/forex/model/RegisterForm.java [hunk 3]
```diff
@@ -3,14 +3,29 @@
...
+
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotNull;
 
 @Data
 @NoArgsConstructor
 @AllArgsConstructor
 public class RegisterForm {
 
+    @NotNull(message = "password is required")
```
Comment (BartlomiejBak) [inline src/main/java/pl/arcadeit/forex/model/RegisterForm.java]:
```suggestion
    @NotNull(message = "email is required")
```

File: src/main/java/pl/arcadeit/forex/domain/User.java [hunk 4]
```diff
@@ -22,22 +23,27 @@
...
     @Column(name = "email")
     private String email;
 
-    @Length(max = 25, message = "First name should contain up to 25 characters.")
+    @NotNull(message = "first name is required")
```
Comment (BartlomiejBak) [inline src/main/java/pl/arcadeit/forex/domain/User.java]:
```suggestion
    @NotNull(message = "{first.name.required}")
```

File: src/main/java/pl/arcadeit/forex/domain/User.java [hunk 5]
```diff
@@ -22,22 +23,27 @@
...
     private String firstName;
 
-    @Length(min = 2, max = 50, message = "Last name should contain 2 to 50 characters.")
+    @NotNull(message = "last name is required")
+    @Length(min = 2, max = 50, message = "last name should contain between 2 and 50 characters")
     @Column(name = "last_name")
     private String lastName;
 
-    @Length(min = 8, message = "Password should contain at least 8 characters.")
+    @NotNull(message = "password is required")
```
Comment (BartlomiejBak) [inline src/main/java/pl/arcadeit/forex/domain/User.java]:
```suggestion
    @NotNull(message = "{password.required}")
```

File: src/main/java/pl/arcadeit/forex/domain/User.java [hunk 6]
```diff
@@ -22,22 +23,27 @@
...
     private String email;
 
-    @Length(max = 25, message = "First name should contain up to 25 characters.")
+    @NotNull(message = "first name is required")
+    @Length(min = 2, max = 25, message = "first name should contain between 2 and 25 characters")
     @Column(name = "first_name")
     private String firstName;
 
-    @Length(min = 2, max = 50, message = "Last name should contain 2 to 50 characters.")
+    @NotNull(message = "last name is required")
```
Comment (BartlomiejBak) [inline src/main/java/pl/arcadeit/forex/domain/User.java]:
```suggestion
    @NotNull(message = "{last.name.required}")
```

File: src/main/java/pl/arcadeit/forex/domain/User.java [hunk 7]
```diff
@@ -22,22 +23,27 @@
...
     @Column(name = "last_name")
     private String lastName;
 
-    @Length(min = 8, message = "Password should contain at least 8 characters.")
+    @NotNull(message = "password is required")
+    @Length(min = 8, max = 64, message = "password should contain between 8 and 64 characters")
     @Column(name = "password")
     private String password;
 
+    @NotNull(message = "role is required")
```
Comment (BartlomiejBak) [inline src/main/java/pl/arcadeit/forex/domain/User.java]:
```suggestion
    @NotNull(message = "{role.invalid}")
```

File: src/main/java/pl/arcadeit/forex/domain/User.java [hunk 8]
```diff
@@ -22,22 +23,27 @@
...
 
-    @Length(min = 2, max = 50, message = "Last name should contain 2 to 50 characters.")
+    @NotNull(message = "last name is required")
+    @Length(min = 2, max = 50, message = "last name should contain between 2 and 50 characters")
     @Column(name = "last_name")
     private String lastName;
 
-    @Length(min = 8, message = "Password should contain at least 8 characters.")
+    @NotNull(message = "password is required")
+    @Length(min = 8, max = 64, message = "password should contain between 8 and 64 characters")
```
Comment (BartlomiejBak) [inline src/main/java/pl/arcadeit/forex/domain/User.java]:
```suggestion
    @Length(min = 8, max = 64, message = "{password.invalid}")
```

File: src/main/java/pl/arcadeit/forex/domain/User.java [hunk 9]
```diff
@@ -22,22 +23,27 @@
...
     @Id
-    @Email(message = "Email address is invalid.")
+    @NotNull(message = "{email.required}")
+    @Email(message = "{email.invalid}")
     @Column(name = "email")
     private String email;
 
-    @Length(max = 25, message = "First name should contain up to 25 characters.")
+    @NotNull(message = "first name is required")
+    @Length(min = 2, max = 25, message = "first name should contain between 2 and 25 characters")
```
Comment (BartlomiejBak) [inline src/main/java/pl/arcadeit/forex/domain/User.java]:
```suggestion
    @Length(min = 2, max = 25, message = "{first.name.invalid}")
```

File: src/main/java/pl/arcadeit/forex/domain/User.java [hunk 10]
```diff
@@ -22,22 +23,27 @@
...
     @Column(name = "first_name")
     private String firstName;
 
-    @Length(min = 2, max = 50, message = "Last name should contain 2 to 50 characters.")
+    @NotNull(message = "last name is required")
+    @Length(min = 2, max = 50, message = "last name should contain between 2 and 50 characters")
```
Comment (BartlomiejBak) [inline src/main/java/pl/arcadeit/forex/domain/User.java]:
```suggestion
    @Length(min = 2, max = 50, message = "{last.name.invalid}")
```

**Review by BartlomiejBak — DISMISSED** [review]
only some minor issues

**Comment by Enessetere** [comment]
I will fix issues reported by You and reopen this pull request

**Comment by Enessetere** [comment]
Please review changes I've made and let me know is there still something that needs changes.

**Review by PrzyblowskiBartosz — APPROVED** [review]

**Review by BartlomiejBak — APPROVED** [review]
looks good for me

**Pull request merged**